### PR TITLE
feat: patch release workflow

### PR DESCRIPTION
**Pull request by ivanmjartan**
JIRA: STL-122

---

> [!IMPORTANT]
> Please, **don't forget to run `rush change`** for the commits that introduce **new features** 🙏

---

Refer to [documentation](https://github.com/gooddata/gooddata-ui-sdk/blob/master/dev_docs/continuous_integration.md) to see how to run checks and tests in the pull request. This is the list of the most used commands:

```
extended test - backstop
```

```
// Tiger platform
extended test - tiger-cypress - integrated
extended test - tiger-cypress - isolated
extended test - tiger-cypress - record
```

```
// Bear platform
extended test - cypress - integrated
extended test - cypress - isolated
extended test - cypress - record
```

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18099/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/19649/ : FAILURE in 1m 37s

File: .github/actions/git-check-release/action.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# (C) 2022 GoodData Corporation
+
+name: "Action to update hugo version param in parmas.toml"
+description: "Action add new version to hugo version config set it latest and remove old one"
```
Comment (pbenes) [inline .github/actions/git-check-release/action.yml:4]: fix the description & name

File: .github/workflows/git-version-check.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: A ~ check latest release version
+on:
+    workflow_dispatch:
+        inputs:
+            version:
+                description: 'Version to backmerge'
+                required: true
+                default: '0.0.0'
```
Comment (pbenes) [inline .github/workflows/git-version-check.yml:8]: I would probably not add default if it's required

File: .github/workflows/prerelease-beta-patch.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        outputs:
+            branch: ${{ steps.check-input-version-param.outputs.branch }}
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@v4
+            - name: Check if branch for version exists
+              id: check-input-version-param
+              run: |
+                  base="rel/${VERSION%.*}"
+                  # TODO: brackets for if?
```
Comment (pbenes) [inline .github/workflows/prerelease-beta-patch.yml:23]: please unify the brackets or just remove TODO

File: .github/workflows/release-release-patch.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        outputs:
+            branch: ${{ steps.check-input-version-param.outputs.branch }}
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@v4
+            - name: Check branch for version exists
+              id: check-input-version-param
+              run: |
+                  base="rel/${VERSION%.*}"
+                  if git ls-remote --heads origin ${base} | grep ${base}; then
```
Comment (pbenes) [inline .github/workflows/release-release-patch.yml:24]: can we use the existing action?

**Comment by ivanmjartan** [comment]
ok to test

File: .github/workflows/rw-publish-prerelease.yml [hunk 5]
```diff
@@ -40,33 +40,47 @@ jobs:
...
     outputs:
       prerelease-id: ${{ steps.validate-params.outputs.prerelease-id }}
-
+      prerelease-tag: ${{ steps.validate-params.outputs.prerelease-tag }}
     steps:
       - name: validate params
         id: validate-params
         run: |
-          if [ "${{ inputs.source-branch }}" = "master" ]; then
+          # TODO: remove inputs from bash script, use env
```
Comment (pbenes) [inline .github/workflows/rw-publish-prerelease.yml:74]: should be easy to fix, can you please move to env

File: .github/actions/git-check-release/action.yml (hunk 1 above)
Comment (ivanmjartan) [reply .github/actions/git-check-release/action.yml:4]: this manual WF needs to be deleted it was there for testing

File: .github/workflows/prerelease-beta-patch.yml (hunk 3 above)
Comment (ivanmjartan) [reply .github/workflows/prerelease-beta-patch.yml:23]: done

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18100/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/19650/ : SUCCESS in 12m 27s

File: .github/workflows/rw-publish-prerelease.yml (hunk 5 above)
Comment (ivanmjartan) [reply .github/workflows/rw-publish-prerelease.yml:74]: done

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/18101/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/19652/ : SUCCESS in 12m 31s

**Review by pbenes — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR4695

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/3927/ : SUCCESS in 8m 54s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/19653/ : SUCCESS in 13m 19s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/1584/ : SUCCESS in 10m 20s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/5095/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/1806/ : FAILURE in 1m 17s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/1594/ : SUCCESS in 12m 30s

**Pull request merged**